apps/wallet/src/ui/app/pages/qredo-connect/QredoConnectInfoPage.tsx

```diff
@@ -1,27 +1,121 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
-import { useParams } from 'react-router-dom';
+import { useEffect, useMemo, useState } from 'react';
+import { useNavigate, useParams } from 'react-router-dom';
 
-import { Button } from '../../shared/ButtonUI';
+import { useBackgroundClient } from '../../hooks/useBackgroundClient';
+import { Heading } from '../../shared/heading';
 import { PageMainLayoutTitle } from '../../shared/page-main-layout/PageMainLayoutTitle';
 import { Text } from '../../shared/text';
+import { useQredoUIPendingRequest } from './hooks';
+import { isUntrustedQredoConnect } from './utils';
+import { LabelValueItem } from '_components/LabelValueItem';
+import { LabelValuesContainer } from '_components/LabelValuesContainer';
+import { SummaryCard } from '_components/SummaryCard';
+import { UserApproveContainer } from '_components/user-approve-container';
 
 export function QredoConnectInfoPage() {
     const { requestID } = useParams();
-    // eslint-disable-next-line no-console
-    console.log('QredoConnectInfoPage', requestID);
+    const { data, isLoading } = useQredoUIPendingRequest(requestID);
+    const isUntrusted = useMemo(
```

Comment:
I don't think we should memo-ize this unless the operation is expensive or we need referential stability